### PR TITLE
Make Qt auto-detect the preferred page size for printer

### PR DESCRIPTION
**Pull request by Anish-M-code**
-> Removes the need for hard-coded paper size in printer configuration if QT 5.3 or above is used related to issue #3152

### CODE REVIEW

**Comment by chromatic** [comment]
This looks generally correct to me; concept ACK.

I wonder if it would read better AND work correctly if there were only one `#define` block; is it possible to get and set paper/page size that way?

**Comment by Anish-M-code** [comment]
should i enclose my changes in a single #define block  and like #define USE_DEFAULT_PAGE_SIZE and set it true to conditionally compile my existing changes.

**Comment by chromatic** [comment]
I tested this and the code compiles and passes tests fine; I need to test again with a printer on my network (it just disappeared during testing).

Printing to PDF suggests a paper size of A4, but I'm not sure if/how this would address that issue. So far so good for the effects.

**Comment by Anish-M-code** [comment]
@chromatic kindly let me know in case if i had to change code if code doesnt work as expected, i usually test code before submitting PR but i didnt have printer setup so i couldnt test this one.

**Comment by Anish-M-code** [comment]
@chromatic kindly let me know if my feature doesn't work as expected, I usually test code before submitting PR but i didnt have printer setup so i couldnt test this one.

**Comment by chromatic** [comment]
I apologize for the delay.

Tested ACK -- code builds and passes all tests. I tried running this code without an active printer on the network, and the paper size defaulted to A4. Then I connected a printer to the network with a different paper size (US Letter by default), and the print dialog used the printer's default.

I'd like to have the commits squashed before merging. If you're willing to do that @Anish-M-code, that'd be great. Otherwise one if the core maintainers can do it for you. Do you have a preference?

**Comment by chromatic** [comment]
I've rebased and squashed, so this could use one more maintainer's review before a final merge.

**Review by patricklodder — APPROVED** [review]
ACK. Tested on Windows 11.

- Paper size selection ✅
- Aspect ratios retained ✅
- Default A4 fallback ✅

Apologies for this taking so long to confirm. I had to do this while visiting family that (a) had supports a printer with multiple paper sizes and (b) uses Windows PCs I could test on.

**Pull request merged**
